File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    canPartition(nums) { 
        let sum = nums.reduce((acc, cur) => acc + cur, 0) 
        if (sum % 2 !== 0) return false 

        return this.dfs(0, nums, 0, sum / 2)
    } 

    dfs(start, nums, cur, target) { 
        if (cur > target) return false
        if (cur === target) return true

        for (let i = start; i < nums.length; i ++) {
            if (this.dfs(i + 1, nums, cur + nums[i], target)) return true
        } 

        return false
    }
}

File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1:
            return True

        dp = [False] * n
        dp[0] = True

        for i in range(n):
            if not dp[i]:
                continue

            for j in range(1, nums[i] + 1):
                if i + j < n:
                    dp[i + j] = True

        return dp[-1]